libsimulator/src/Point.cpp

```diff
@@ -1,12 +1,20 @@
 // Copyright © 2012-2024 Forschungszentrum Jülich GmbH
 // SPDX-License-Identifier: LGPL-3.0-or-later
 #include "Point.hpp"
-
-#include "Macros.hpp"
+#include <cassert>
 
 #include <Logger.hpp>
 #include <limits>
 
+bool Point::isInvalidOrientation() const
+{
+    if((std::abs(x) < 1e-6 && std::abs(y) < 1e-6) || (std::abs(NormSquare() - 1.0) > 1e-6)) {
+        LOG_WARNING("Orientation vector ({}, {}) is not normalized, correcting it.", x, y);
+        return true;
+    }
+    return false;
+}
+
```

Comment:
I think this should be renamed to 'isZeroLength' because the current name implies that the Point has to be used as orientation for this function to make sense.

Also the warning message is misleading because the no correction is taking place and in general this is not something we should "warn" the user about. For the warning we are operating on at least one abstraction level below the code that should log a warning, if at all.